src/server/cluster/cluster_family.cc

```diff
@@ -45,7 +45,7 @@ constexpr string_view kClusterDisabled =
     "Cluster is disabled. Enabled via passing --cluster_mode=emulated|yes";
 constexpr string_view kDflyClusterCmdPort = "DflyCluster command allowed only under admin port";
 
-thread_local unique_ptr<ClusterConfig> tl_cluster_config;
+thread_local shared_ptr<ClusterConfig> tl_cluster_config;
```

Comment:
I find it suspicious that the only thing you had to change is the type.
Can you point to a place where shared_ptr semantics now make a difference?

Reply:
The only difference should be upon assignment or construction/destruction. Now a single instance will be in memory instead of a copy per thread.
What could make your suspicions go away? :laughing: I'm happy to add tests / prove otherwise

Reply:
Oh, I missed that you copied by value before 🤦🏼 I thought you moved unique_ptr somewhere...

LGTM!